### PR TITLE
arch/sim: improve arch text heap implementation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- arch/sim: Call host_memalign in up_allocate_heap 
- arch/sim: Simplify the implementation of textheap

## Impact

sim elf loader

## Testing

sim:elf

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Comment by yamt** [comment]
what's your motivation?

.
as you see in the CI results, this affects more things than sim.

**Comment by xiaoxiang781216** [comment]
> what's your motivation?
> 
> > arch: Change ARCH_USE_TEXT_HEAP default to y instead selection
> 
> as you see in the CI results, this affects more things than sim. in case of esp32, it also controls associated trap handler behavior.

since it's impossible to disable ARCH_USE_TEXT_HEAP from defconfig, if selection is done in Kconfig.

File: arch/sim/src/sim/posix/sim_hostmemory.c [hunk 1]
```diff
@@ -60,18 +60,18 @@ static atomic_int g_uordblks;
...
   /* see: https://developer.apple.com/forums/thread/672804 */
 
-  p = host_uninterruptible(mmap, NULL, sz, PROT_READ | PROT_WRITE,
+  p = host_uninterruptible(mmap, NULL, size, PROT_READ | PROT_WRITE,
                            MAP_ANON | MAP_SHARED, -1, 0);
 #else
-  p = host_uninterruptible(mmap, NULL, sz,
-                           PROT_READ | PROT_WRITE | PROT_EXEC,
+  p = host_uninterruptible(mmap, NULL, size, PROT_READ | PROT_WRITE |
+                           exec ? PROT_EXEC : 0,
```
Comment (Donny9) [inline arch/sim/src/sim/posix/sim_hostmemory.c:74]: need to add () to exec ? PROT_EXEC : 0

Comment (xiaoxiang781216) [reply arch/sim/src/sim/posix/sim_hostmemory.c]: Done.

**Comment by xiaoxiang781216** [comment]
@yamt the patch you concern is removed from pr, please review again.

**Pull request closed without merging**